backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/network/host/HostSetupNetworksCommand.java

```diff
@@ -340,26 +340,43 @@ protected void executeCommand() {
 
         try (EngineLock monitoringLock = acquireMonitorLock("Host setup networks")) {
             int timeout = getSetupNetworksTimeout();
-            FutureVDSCall<VDSReturnValue> setupNetworksTask = invokeSetupNetworksCommand(timeout);
 
             try {
-                VDSReturnValue retVal = setupNetworksTask.get(timeout, TimeUnit.SECONDS);
-                if (retVal != null) {
-                    if (!retVal.getSucceeded() && retVal.getVdsError() == null && getParameters().rollbackOnFailure()) {
-                        throw new EngineException(EngineError.SETUP_NETWORKS_ROLLBACK, retVal.getExceptionString());
+                boolean networksUpdated = false;
```

Comment:
We need only single boolean. The fact that we have propagated succeed even if `retVal.getSucceeded() == false` seems to be a mistake. That was introduced by 28f34a48.

Reply:
@almusil this significantly simplifies the proposed changes. Thanks a lot! Will do it soon - it may take a while to verify the changes.

Reply:
Fixed